diagnostics/app/model/diagnostics/commercial/CommercialReport.scala

```diff
@@ -0,0 +1,49 @@
+package model.diagnostics.commercial
+
+import play.api.libs.json.{Json, JsSuccess, JsValue}
+
+case class Module(
+  duration: Double,
+  name: String,
+  start: Double )
+
+case class Advert(
+  createTime: Option[Double],
+  dfpFetching: Option[Double],
+  dfpReceived: Option[Double],
+  dfpRendered: Option[Double],
+  id: Option[String],
+  isEmpty: Option[String],
+  lazyWaitComplete: Option[Double],
+  loadingMethod: Option[String],
+  startLoading: Option[Double],
+  startRendering: Option[Double],
+  stopLoading: Option[Double],
+  stopRendering: Option[Double]
+)
+
+case class Baseline(
+  name: Option[String],
+  time: Option[Double]
+)
+
+case class Report(
+  viewId: String,
+  modules: Seq[Module],
+  adverts: Seq[Advert],
+  baselines: Seq[Baseline])
+
+object Report extends common.Logging {
+
+  implicit val moduleFormat = Json.format[Module]
+  implicit val advertFormat = Json.format[Advert]
+  implicit val baselineFormat = Json.format[Baseline]
+  implicit val reportFormat = Json.format[Report]
+
+  def report(requestBody: JsValue): Unit = {
+    requestBody.validate[Report] match {
+      case JsSuccess(report, _) => RedisReport.report(report)
+      case _ =>
```

Comment:
Get rid of the underscore and use `JsError` instead to avoid having a wildcard.. Also maybe we want logging here? If it falls into this case you are going to have to end up adding logging anyway!